tutorials/developing_with_ign-cmake.md

```diff
@@ -0,0 +1,197 @@
+\page developingwithcmake Developing with Ignition CMake 
+
+# Developing with Ignition CMake
+
+This tutorial documents various tips and strategies for developing with Ignition CMake.
+
+## Helpful CMake flags
+
+There are several flags that control the results of the CMake tool.
+Some of these flags are built into CMake, where some are Ignition CMake specific.
+
+### Setting the build type
+
+The `CMAKE_BUILD_TYPE` variable controls the type of binary output from the build stage.
+This will have an impact on the flags passed to the compiler and linker.
+
+The available options are:
+
+  * `RelWithDebInfo` (default): Mostly optimized build, but with debug symbols.
+  * `Debug`: Debug build without optimizations, with debug symbols enabled.
+  * `Release`: Fully optimized build, no debug symbols.
+  * `MinSizeRel`: Fully optimized build, minimal binary size 
+  * `Coverage`: Build with additional information required for code coverage computation
+  * `Profile`: Use flags that are helpful with the `gprof` profiling tool
+
+More information about flags applied can be found in [IgnSetCompilerFlags.cmake](https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/cmake/IgnSetCompilerFlags.cmake)
+
+### Creating a compilation database
+
+`CMake` can optionally generate a compilation data base that may be used with a variety of code completion tools.
+To enable this set: `-DCMAKE_EXPORT_COMPILE_COMMANDS=ON`.
+
+### Using the ninja build system
+
+Rather than using `make`, it may be desired to use the [Ninja](https://ninja-build.org/) build tool.
+
+This may be applied by setting `-GNinja`.
+
+### Address sanitizer
+
+The `gcc` and `clang` compilers have a set of flags to generate instrumented builds for detecting memory leaks.
+
+```
+-DCMAKE_CXX_FLAGS:STRING= -fsanitize=address  -fsanitize=leak -g
+-DCMAKE_C_FLAGS:STRING=-fsanitize=address  -fsanitize=leak -g
+-DCMAKE_EXE_LINKER_FLAGS:STRING=-fsanitize=address  -fsanitize=leak
+-DCMAKE_MODULE_LINKER_FLAGS:STRING=-fsanitize=address  -fsanitize=leak
```

Comment:
A few questions came to my mind when reading through this. Perhaps we can address them in the tutorial:
1. What are the differences between all of these flags?
2. Should users use all of these flags to check for memory leaks, or can they pick and choose particular flags based on their needs?

Reply:
All of the flags are required for address and leak checking.  I updated the wording to indicate that.